### PR TITLE
[Spark] Move VacuumProtocolCheck Table Feature from Dev/Testing to Prod

### PR DESCRIPTION
**Pull request by sumeet-db**
#### Which Delta project/connector is this regarding?
- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description
We move the Table Feature from dev/test only to prod as part of this PR.

## How was this patch tested?
Existing UTs

## Does this PR introduce _any_ user-facing changes?
No

### CODE REVIEW

File: spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala [hunk 1]
```diff
@@ -335,6 +335,7 @@ object TableFeature {
       IcebergCompatV1TableFeature,
       IcebergCompatV2TableFeature,
       DeletionVectorsTableFeature,
+      VacuumProtocolCheckTableFeature,
```
Comment (prakharjain09) [inline spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala:338]: As part of this change, we should also finalize the vacuumProtocolCheck RFC and update the Delta spec.

Comment (sumeet-db) [reply spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala:338]: Added it in a separate PR https://github.com/delta-io/delta/pull/2808/files

**Review by prakharjain09 — APPROVED** [review]

File: spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala [hunk 2]
```diff
@@ -682,7 +682,7 @@ object InCommitTimestampTableFeature
  * feature exists
  */
 object VacuumProtocolCheckTableFeature
-  extends ReaderWriterFeature(name = "vacuumProtocolCheck-dev")
+  extends ReaderWriterFeature(name = "vacuumProtocolCheck")
```
Comment (tdas) [inline spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala:685]: isnt there a change needed in the tests to make sure that you dont have to enable some testing-only internal flag to use this feature?

Comment (sumeet-db) [reply spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala:685]: The change to ensure the feature is available without enabling a testing-only internal flag is included in the current file by moving it from testing-set to prod-set. 

```
if (DeltaUtils.isTesting) {
      features ++= Set(
        TestLegacyWriterFeature, ....
```

Comment (tdas) [reply spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala:685]: I see. thats how it works. good.

**Review by tdas — APPROVED** [review]

**Pull request merged**